hazelcast/src/test/java/com/hazelcast/replicatedmap/merge/ReplicatedMapSplitBrainTest.java

```diff
@@ -0,0 +1,251 @@
+    protected void onAfterSplitBrainCreated(HazelcastInstance[] firstBrain, HazelcastInstance[] secondBrain) {
+        mergeLifecycleListener = new MergeLifecycleListener(secondBrain.length);
+        for (HazelcastInstance instance : secondBrain) {
+            instance.getLifecycleService().addLifecycleListener(mergeLifecycleListener);
+        }
+
+        replicatedMap1 = firstBrain[0].getReplicatedMap(replicatedMapName);
+        replicatedMap2 = secondBrain[0].getReplicatedMap(replicatedMapName);
+
+        if (mergePolicyClass == LatestUpdateMapMergePolicy.class) {
+            afterSplitLatestUpdateMapMergePolicy();
+        }
+        if (mergePolicyClass == HigherHitsMapMergePolicy.class) {
+            afterSplitHigherHitsMapMergePolicy();
+        }
+        if (mergePolicyClass == PutIfAbsentMapMergePolicy.class) {
+            afterSplitPutIfAbsentMapMergePolicy();
+        }
+        if (mergePolicyClass == PassThroughMergePolicy.class) {
+            afterSplitPassThroughMapMergePolicy();
+        }
+        if (mergePolicyClass == CustomReplicatedMergePolicy.class) {
+            afterSplitCustomReplicatedMapMergePolicy();
+        }
+    }
+
+    @Override
+    protected void onAfterSplitBrainHealed(HazelcastInstance[] instances) {
+        // wait until merge completes
+        mergeLifecycleListener.await();
+
+        if (mergePolicyClass == LatestUpdateMapMergePolicy.class) {
+            afterMergeLatestUpdateMapMergePolicy();
+        }
+        if (mergePolicyClass == HigherHitsMapMergePolicy.class) {
+            afterMergeHigherHitsMapMergePolicy();
+        }
+        if (mergePolicyClass == PutIfAbsentMapMergePolicy.class) {
+            afterMergePutIfAbsentMapMergePolicy();
+        }
+        if (mergePolicyClass == PassThroughMergePolicy.class) {
+            afterMergePassThroughMapMergePolicy();
+        }
+        if (mergePolicyClass == CustomReplicatedMergePolicy.class) {
+            afterMergeCustomReplicatedMapMergePolicy();
+        }
+    }
+
+    private void afterSplitLatestUpdateMapMergePolicy() {
+        replicatedMap1.put("key1", "value1");
+
+        // prevent updating at the same time
+        sleepAtLeastMillis(100);
+
+        replicatedMap2.put("key1", "LatestUpdatedValue1");
+        replicatedMap2.put("key2", "value2");
+
+        // prevent updating at the same time
+        sleepAtLeastMillis(100);
+
+        replicatedMap1.put("key2", "LatestUpdatedValue2");
+    }
+
+    private void afterMergeLatestUpdateMapMergePolicy() {
+        assertEquals("LatestUpdatedValue1", replicatedMap1.get("key1"));
+        assertEquals("LatestUpdatedValue1", replicatedMap2.get("key1"));
+
+        assertEquals("LatestUpdatedValue2", replicatedMap1.get("key2"));
+        assertEquals("LatestUpdatedValue2", replicatedMap2.get("key2"));
+    }
+
+    private void afterSplitHigherHitsMapMergePolicy() {
+        // hits are not replicated and both nodes of the larger cluster can be the merge target,
```

Comment:
This scenario gave me a little headache (since it was failing randomly, when just `replicatedMap1` was used), but I think my assumption is correct and the solution works smoothly.